constantinCurtis/src/main/java/data/collection/task/classes/AbstractOrder.java

```diff
@@ -0,0 +1,104 @@
+package data.collection.task.classes;
+
+import data.collection.task.Interfaces.Order;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+public abstract class AbstractOrder implements Order {
+    private final String id;
+    private final LocalDateTime date;
+    private List<Item> items;
+
+
+    public AbstractOrder(List<Item> items) {
+        this.id = randomUUID();
+        this.date = LocalDateTime.now();
+        this.items = items;
+    }
+
+    private String randomUUID() {
+        return String.valueOf(UUID.randomUUID());
+    }
+
+    @Override
+    public int count() {
+        return this.items.size();
+    }
+
+    public int count(Item item) {
+        int itemSize = 0;
+        for (Item entity : this.items) {
+            itemSize++;
+        }
+        return itemSize;
+    }
```

Comment:
I suggest to count only items corresponding to given as parameter item title. Otherwise "item" parameter is redundant.

Reply:
done